pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -19,35 +20,60 @@ import (
 
 var log = logf.Log.WithName("templates")
 
+const (
+	// ClusterResources the key to retrieve the cluster resources template
+	ClusterResources string = "clusterResources" // TODO: move in API and remove duplicate in https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/nstemplateset/nstemplatetier.go#L18
+)
+
 // CreateOrUpdateResources generates the NSTemplateTier resources from the namespace templates,
 // then uses the manager's client to create or update the resources on the cluster.
 func CreateOrUpdateResources(s *runtime.Scheme, client client.Client, namespace string, assets Assets) error {
 	templatesByTier, err := loadTemplatesByTiers(assets)
 	if err != nil {
 		return errors.Wrap(err, "unable to create or update NSTemplateTiers")
 	}
-	tierObjs, err := newNSTemplateTiers(s, namespace, templatesByTier)
+	decoder := serializer.NewCodecFactory(s).UniversalDeserializer()
+	cl := commonclient.NewApplyClient(client, s)
+
+	// create the TierTemplates
+	tierTmpls, err := newTierTemplates(decoder, namespace, templatesByTier)
+	if err != nil {
+		return errors.Wrap(err, "unable to create or update TierTemlates")
+	}
+	for _, tierTmpl := range tierTmpls {
+		log.Info("creating or updating TierTemplate", "namespace", tierTmpl.Namespace, "name", tierTmpl.Name)
+		createdOrUpdated, err := cl.CreateOrUpdateObject(tierTmpl, true, nil)
+		if err != nil {
+			return errors.Wrapf(err, "unable to create or update the '%s' TierTemplate in namespace '%s'", tierTmpl.Name, tierTmpl.Namespace)
+		}
+		if createdOrUpdated {
+			log.Info("NSTemplateTier resource created/updated", "namespace", tierTmpl.Namespace, "name", tierTmpl.Name)
+		} else {
+			log.Info("NSTemplateTier resource was already up-to-date", "namespace", tierTmpl.Namespace, "name", tierTmpl.Name, "ResourceVersion", tierTmpl.ResourceVersion)
+		}
```

Comment:
Yeah... Maybe the Shane's idea of naming it `TierTemplateRevision` instead of `TierTemplate` would make it clearer that it's just a revision/snapshot/version of the template, so, it's kinda immutable.

Reply:
right, let me change that.

Reply:
done in c3ce432